Formats/FileNamer.cpp

```diff
@@ -96,16 +129,27 @@ std::string FileNamer::filename( const std::unique_ptr<SignalSet>& data ) {
     "%s",
     "%t",
     "%o",
-    "%C"
+    "%C",
+    "%D",
+    "%e",
+    "%x",
+    "%m",
+    "%S"
   };
-
+  size_t pos = lastname.find ("%S");
```

Comment:
Small thing, but please do the %S replacement(s) in the `parse()` method, just before the return. It's a one-time deal, so no reason to do it every time a filename is generated.

Reply:
Sure thing! Anything else need an adjustment while I'm doing another commit?

Reply:
I put %S in parse, but I since it treated pattern a little differently than filename() I wasn't sure if copying and returning a new string was the best call. It seems to run fine, but I don't have an example to run it on.